l1-contracts/contracts/bridge/asset-router/L1AssetRouter.sol

```diff
@@ -0,0 +1,524 @@
+    {
+        _disableInitializers();
+        L1_WETH_TOKEN = _l1WethAddress;
+        ERA_DIAMOND_PROXY = _eraDiamondProxy;
+        L1_NULLIFIER = IL1Nullifier(_l1Nullifier);
+    }
+
+    /// @dev Initializes a contract bridge for later use. Expected to be used in the proxy.
+    /// @dev Used for testing purposes only, as the contract has been initialized on mainnet.
+    /// @param _owner The address which can change L2 token implementation and upgrade the bridge implementation.
+    /// The owner is the Governor and separate from the ProxyAdmin from now on, so that the Governor can call the bridge.
+    function initialize(address _owner) external reentrancyGuardInitializer initializer {
+        if (_owner == address(0)) {
+            revert ZeroAddress();
+        }
+        _transferOwnership(_owner);
+    }
+
+    /// @notice Sets the L1ERC20Bridge contract address.
+    /// @dev Should be called only once by the owner.
+    /// @param _nativeTokenVault The address of the native token vault.
+    function setNativeTokenVault(INativeTokenVault _nativeTokenVault) external onlyOwner {
+        require(address(nativeTokenVault) == address(0), "AR: native token v already set");
+        require(address(_nativeTokenVault) != address(0), "AR: native token vault 0");
+        nativeTokenVault = _nativeTokenVault;
+    }
+
+    /// @notice Sets the L1ERC20Bridge contract address.
+    /// @dev Should be called only once by the owner.
+    /// @param _legacyBridge The address of the legacy bridge.
+    function setL1Erc20Bridge(address _legacyBridge) external onlyOwner {
+        if (address(legacyBridge) != address(0)) {
+            revert AddressAlreadyUsed(address(legacyBridge));
+        }
+        if (_legacyBridge == address(0)) {
+            revert ZeroAddress();
+        }
+        legacyBridge = _legacyBridge;
+    }
+
+    /// @notice Used to set the assed deployment tracker address for given asset data.
+    /// @param _assetRegistrationData The asset data which may include the asset address and any additional required data or encodings.
+    /// @param _assetDeploymentTracker The whitelisted address of asset deployment tracker for provided asset.
+    function setAssetDeploymentTracker(
+        bytes32 _assetRegistrationData,
+        address _assetDeploymentTracker
+    ) external onlyOwner {
+        bytes32 assetId = keccak256(
+            abi.encode(uint256(block.chainid), _assetDeploymentTracker, _assetRegistrationData)
+        );
+        assetDeploymentTracker[assetId] = _assetDeploymentTracker;
+        emit AssetDeploymentTrackerSet(assetId, _assetDeploymentTracker, _assetRegistrationData);
+    }
+
+    ///  @inheritdoc IL1AssetRouter
+    function setAssetHandlerAddress(
+        address _prevMsgSender,
+        bytes32 _assetId,
+        address _assetAddress
+    ) external onlyNullifierWithSender(L2_ASSET_ROUTER_ADDR, _prevMsgSender) {
```

Comment:
Because this is not setAssetHandlerAddressInitial, or Counterpart. This is the cross chain version, i.e. the one that is called by Counterpart. This means it will be called from the L2.